tests/blas_test.cpp

```diff
@@ -41,102 +65,146 @@ void display_test_info()
+  caxpyXmazNormX,
+  cabxpyzAxNorm,
+  cDotProduct,
+  caxpyDotzy,
+  cDotProductNormA,
+  caxpbypzYmbwcDotProductUYNormY,
+  HeavyQuarkResidualNorm,
+  xpyHeavyQuarkResidualNorm,
+  tripleCGReduction,
+  tripleCGUpdate,
+  axpyReDot,
+  caxpyBxpz,
+  caxpyBzpx,
+  axpy_block,
+  caxpy_block,
+  axpyBzpcx_block,
+  reDotProductNorm_block,
+  reDotProduct_block,
+  cDotProductNorm_block,
+  cDotProduct_block
+};
 
 // For googletest names must be non-empty, unique, and may only contain ASCII
 // alphanumeric characters or underscore
-const std::vector<std::string> names = {"copyHS",
-                                        "copyMS",
-                                        "copyLS",
-                                        "axpby",
-                                        "xpy",
-                                        "axpy",
-                                        "xpay",
-                                        "mxpy",
-                                        "ax",
-                                        "caxpy",
-                                        "caxpby",
-                                        "cxpaypbz",
-                                        "axpyBzpcx",
-                                        "axpyZpbx",
-                                        "caxpbypzYmbw",
-                                        "cabxpyAx",
-                                        "caxpyXmaz",
-                                        "norm",
-                                        "reDotProduct",
-                                        "axpyNorm",
-                                        "xmyNorm",
-                                        "caxpyNorm",
-                                        "caxpyXmazNormX",
-                                        "cabxpyzAxNorm",
-                                        "cDotProduct",
-                                        "caxpyDotzy",
-                                        "cDotProductNormA",
-                                        "cDotProductNormB",
-                                        "caxpbypzYmbwcDotProductUYNormY",
-                                        "HeavyQuarkResidualNorm",
-                                        "xpyHeavyQuarkResidualNorm",
-                                        "tripleCGReduction",
-                                        "tripleCGUpdate",
-                                        "axpyReDot",
-                                        "caxpy_block",
-                                        "axpyBzpcx_block",
-                                        "caxpyBxpz",
-                                        "caxpyBzpx",
-                                        "cDotProductNorm_block",
-                                        "cDotProduct_block",
-                                        "reDotProductNorm_block",
-                                        "reDotProduct_block",
-                                        "axpy_block"};
+const std::map<Kernel, std::string> kernel_map
```

Comment:
We already do have some maps like that for CLI11, I need to look up what. Maybe we can reuse part of that.

Reply:
I just had a quick look at this.  I don't think CLI11 quite does what we want here.  The `CLI::TransformPairs` is a map that maps a string to an enum.  We need the other way round here, e.g., map an enum to a string.  Perhaps it could be be reused to do what we want, but for now I think sticking with the `std::map` is fine.

Reply:
Agree, just want to avoid that we need to maintain multiple copies of the same maps.